internal/helmdeployer/helmcache/secret.go

```diff
@@ -0,0 +1,76 @@
+package helmcache
+
+import (
+	"context"
+
+	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretClient struct {
+	cache     corecontrollers.SecretCache
+	client    kubernetes.Interface
+	namespace string
+}
+
```

Comment:
This is implementing https://pkg.go.dev/helm.sh/helm/v3@v3.12.3/pkg/storage/driver#Driver right? Can we add something make it obvious and get early errors if we break the interface?

```
var _ storage.Driver = &SecretClient{}
```

Reply:
Oops, it implements https://github.com/helm/helm/blob/v3.12.3/pkg/storage/driver/secrets.go#L50

Reply:
it would break [here](https://github.com/raulcabello/fleet/blob/helm-cache/internal/helmdeployer/deployer.go#L774) if we break the interface. I don't think we need this extra check or https://github.com/rancher/fleet/pull/1809#discussion_r1338480259 . In my opinion that would be against the implicit interfaces in go

Reply:
Yeah true, since the struct is only created to implement the one interface I thought it might be useful. But as you point out, it will report errors when we use it.  I just read https://go.dev/doc/effective_go#interfaces and it says:

>  The appearance of the blank identifier in this construct indicates that the declaration exists only for the type checking, not to create a variable. Don't do this for every type that satisfies an interface, though. By convention, such declarations are only used when there are no static conversions already present in the code, which is a rare event.